queue/mapper_test.go

```diff
@@ -7,160 +7,204 @@ import (
+			fullExporter: false,
+			msg: kafka.FTMessage{
+				Headers: map[string]string{"X-Request-Id": "tid_1234"},
+				Body:    generateRequestBody("http://upp-content-validator.svc.ft.com/content/811e0591-5c71-4457-b8eb-8c22cf093117", map[string]interface{}{}),
+			},
+			expectedNotification: nil,
+		},
+		{
+			name:         "Test that content type of unexpected type will skip mapping",
+			fullExporter: false,
+			msg: kafka.FTMessage{
+				Headers: map[string]string{"X-Request-Id": "tid_1234"},
+				Body: generateRequestBody("http://upp-content-validator.svc.ft.com/content/811e0591-5c71-4457-b8eb-8c22cf093117", map[string]interface{}{
+					"type": 56,
+				}),
+			},
+			expectedNotification: nil,
+		},
+		{
+			name:         "Test that canBeDistributed with value different than yes will skip mapping",
+			fullExporter: false,
+			msg: kafka.FTMessage{
+				Headers: map[string]string{"X-Request-Id": "tid_1234"},
+				Body: generateRequestBody("http://upp-content-validator.svc.ft.com/content/811e0591-5c71-4457-b8eb-8c22cf093117", map[string]interface{}{
+					"type":             "Article",
+					"canBeDistributed": "no",
+				}),
+			},
+			expectedNotification: nil,
+		},
+		{
+			name:         "Test that valid message will map to valid notification",
+			fullExporter: false,
+			msg: kafka.FTMessage{
+				Headers: map[string]string{"X-Request-Id": "tid_1234"},
+				Body: generateRequestBody("http://upp-content-validator.svc.ft.com/content/811e0591-5c71-4457-b8eb-8c22cf093117", map[string]interface{}{
+					"type":             "Article",
+					"canBeDistributed": "yes",
+				}),
+			},
+			expectedNotification: &Notification{
+				Tid:    "tid_1234",
+				EvType: UPDATE,
+				Stub: content.Stub{
+					UUID:        "811e0591-5c71-4457-b8eb-8c22cf093117",
+					ContentType: "Article",
+				},
+			},
+		},
+		{
+			name:         "Test that valid message will map to valid notification - delete event",
+			fullExporter: false,
+			msg: kafka.FTMessage{
+				Headers: map[string]string{"X-Request-Id": "tid_1234"},
+				Body: generateRequestBody("http://upp-content-validator.svc.ft.com/content/811e0591-5c71-4457-b8eb-8c22cf093117", map[string]interface{}{
+					"type":             "Article",
+					"canBeDistributed": "yes",
+					"deleted":          true,
+				}),
+			},
+			expectedNotification: &Notification{
+				Tid:    "tid_1234",
+				EvType: DELETE,
+				Stub: content.Stub{
+					UUID:        "811e0591-5c71-4457-b8eb-8c22cf093117",
+					ContentType: "Article",
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			regex := exporterRegex
+			if test.fullExporter {
+				regex = fullExporterRegex
+			}
+			mapper := KafkaMessageMapper{
+				regexp.MustCompile(regex),
+				test.fullExporter,
+			}
+			n, err := mapper.MapNotification(test.msg)
+
+			if test.error != "" {
+				if err == nil {
+					t.Fatal("Expected mapping error, got nil")
+				}
+				assert.Contains(t, err.Error(), test.error)
```

Comment:
Can be rewritten as:

```suggestion
                                require.EqualError(t, err, test.error)
```